apple-codesign/src/notarization.rs

```diff
@@ -464,9 +464,17 @@ pub fn write_flat_package_to_app_store_package<F: Read + Seek + Debug>(
         id
     } else if let Some(_component) = pkg.root_component()? {
         warn!("notarizing a component installer");
+        
+        // We need to extract the primary component identifier from the `PackageInfo` XML.
+        let id = match _component.package_info() {
```

Comment:
Nit: `_component` should be renamed to `component` a few lines above.

I'll fix this in flight.